lib/puppet/util/network_device/cisco_ios/device.rb

```diff
@@ -23,10 +20,10 @@ class ModeState
     CONF_EXT_NACL = 14 unless defined? CONF_EXT_NACL
   end
 
-  # Our fun happens here
-  class Puppet::Util::NetworkDevice::Cisco_ios::Device # rubocop:disable Style/ClassAndModuleCamelCase
+  # The main class for handling the connection and command parsing to the IOS Catalyst device
+  class Device < Puppet::Util::NetworkDevice::Simple::Device
     attr_reader :connection
-    attr_accessor :url, :transport, :facts, :commands
+    attr_accessor :transport, :facts, :commands
```

Comment:
Removal of url accessor looks like it may impact: https://github.com/puppetlabs/cisco_ios/blob/00e6a9a6f9bd51c1cca280d67eba1f278d6122e9/tasks/config_save.rb#L84